Cull appview v2 down to frontend service

devinivy opened the pull request:
Reduces the appview down to only use the dataplane service, e.g. dropping the direct dependency on postgres.  Removes all functionality not related to the appview frontends, i.e. indexing, ingestion, and the daemon.  Some indexing functionality has been preserved in `src/data-plane/server` which is a postgres implementation of the dataplane.

dholms commented on packages/bsky/tests/auto-moderator/fuzzy-matcher.test.ts:
```diff
@@ -1,165 +0,0 @@
-import { AtpAgent } from '@atproto/api'
-import { ImageInvalidator } from '../../src/image/invalidator'
-
-describe('fuzzy matcher', () => {
```
hmmm ripping out automod stuff is definitely going to clash with my work, but hopefully by the time we're landing this branch, the separate automod service is in a good enough spot that it's alright 👌

devinivy replied:
In my view we're ripping everything out in this PR to match the end state we want for the appview frontends.  It's hard to imagine how automod functionality would remain here in v2, so I think we nix it and just handle the merge conflict in this branch at the next opportunity, even if it might be a little while.

dholms commented on packages/bsky/tests/db.test.ts:
```diff
@@ -1,10 +1,7 @@
-import { Leader } from '../src/db/leader'
+import { Database, PrimaryDatabase } from '../src'
 
 describe('db', () => {
```
since this is only used for mocks, you can probably just delete this whole test 🤷‍♂️

dholms replied:
actually same with several of these - did-cache, duplicate-records, handle-invalidation, etc

devinivy replied:
Sort of torn on this!  If the code is still around and is important for correctness of the other tests, maybe it's worth keeping them around in some form.  Pretty sure it will remain low cost to maintain.  Perhaps they could just be collapsed into a single file and condensed a bit.

dholms commented on packages/bsky/tests/views/__snapshots__/notifications.test.ts.snap:
```diff
@@ -245,24 +245,7 @@ Array [
-        "val": "test-label-2",
-      },
-    ],
+    "labels": Array [],
```
hmmm these were good integration tests that labels were being hydrated correctly - think we could just add these labels directly in the seed scripts?

not a _huge_ deal, but we don't currently have separate tests for label hydration 😬

devinivy replied:
Seems like a good idea!  Will take a look at that.

dholms commented on packages/pds/tests/proxied/feedgen.test.ts:
```diff
@@ -19,11 +18,16 @@ describe('feedgen proxy view', () => {
       dbPostgresSchema: 'proxy_feedgen',
-      bsky: { algos: makeAlgos(feedUri.host) },
+      bsky: {
+        // @TODO consider using makeAlgos() here if the appview begins supporting any feeds out of the box
```
i might actually just cull all the tests related to this

dholms commented on packages/bsky/tests/notification-server.test.ts:
```diff
@@ -1,231 +0,0 @@
-import { NotificationServer } from '../src/notifications'
-import { Database } from '../src'
-
-describe('notification server', () => {
```
any idea the future of this stuff? it'll be handled by the dataplane indexer i guess? should just make a note so we don't forget about it 😅

devinivy replied:
Indeedy, looking like it will move into the indexer.

dholms reviewed (commented):
High level comment, but I also think we can clean up a lot of unused stuff from the database now that it's just used for mocks (such as the database coordinator, background queue for indexing, etc). 

But also we probably want to eventually just convert this to sqlite, so I don't think it's necessary to take a pass on all that right now

dholms reviewed (approved):
Looks good! & stoked to get to cut all this out 🕺 

I wasn't _super_ thorough in my review, but I scanned the whole PR & grabbed the branch to poke around through the important bits.

The only big thing I think I'll miss is label hydration in all the snaps, it'd be great to find another way to get that in there if possible.

Besides that, I think there's just other code that can be culled as well, but it's not super pressing & we can probably save that for a clean up pass once it's all said & done

devinivy commented:
@dholms made some changes!
 - got those labels back into the tests.
 - moved dataplane server-related tests out of the way into their own directory.
 - removed primary/replica/coordinator junk from the dataplane db.
